src/sql/base/browser/ui/selectBox/selectBox.ts

```diff
@@ -60,20 +64,30 @@ export class SelectBox extends vsSelectBox {
 
 	private element?: HTMLElement;
 
-	constructor(options: string[], selectedOption: string, contextViewProvider: IContextViewProvider, container?: HTMLElement, selectBoxOptions?: ISelectBoxOptions) {
-		super(options.map(option => { return { text: option }; }), 0, contextViewProvider, undefined, selectBoxOptions);
+	constructor(options: SelectOptionItemSQL[] | string[], selectedOption: string, contextViewProvider: IContextViewProvider, container?: HTMLElement, selectBoxOptions?: ISelectBoxOptions) {
+		let optionItems: SelectOptionItemSQL[];
+		if (options.length > 0 && typeof options[0] === 'string') {
```

Comment:
I don't think you can typecheck like that on runtime.

Reply:
https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates

Using type predicate functions allows Typescript to do type narrowing - so it'll know that since it isn't a string[] that it must be a SelectOptionItemSQL[] so you can just do `optionItems = options;` on line 74

Reply:
Eh I guess I'm not handling the situation where the option is a non string type so it wouldn't hurt. Hmm

Reply:
I mostly meant it won't actually be doing any type checking, but what you're saying makes sense.